src/VisualStudio/CSharp/Impl/CSharpPackage.cs

```diff
@@ -111,11 +117,11 @@ private void RegisterObjectBrowserLibraryManager()
             }
         }
 
-        private void UnregisterObjectBrowserLibraryManager()
+        private async Task UnregisterObjectBrowserLibraryManagerAsync()
         {
             if (_libraryManagerCookie != 0)
             {
-                if (this.GetService(typeof(SVsObjectManager)) is IVsObjectManager2 objectManager)
+                if (await GetServiceAsync(typeof(SVsObjectManager)).ConfigureAwait(true) is IVsObjectManager2 objectManager)
```

Comment:
If `SVsObjectManager` is an STA object, you should probably add a `SwitchToMainThreadAsync` call to this method (and your other async methods that use it) in order to be consistent with the 1st threading rule.